packages/core/utils/src/PromiseQueue.js

```diff
@@ -21,8 +21,19 @@ export default class PromiseQueue {
     this._maxConcurrent = opts.maxConcurrent;
   }
 
-  add(fn: () => Promise<mixed>): void {
-    this._queue.push(fn);
+  add<T>(fn: () => Promise<T>): Promise<T> {
+    return new Promise((resolve, reject) => {
```

Comment:
what's this wrapper for?

Reply:
So that you have access to the results of the promise returned from the function added to the queue. I'm keeping track of in progress request promises so that if that if a an identical request comes in while one is already in progress, it can use the in progress promise. It's not totally needed now as identical dep and asset requests shouldn't fire at the same time, but config request and dev dep requests will bring that possibility. In my config invalidation spike branch `incompleteNodes` turned into `inProgress` promises so I just decided to bring it in now instead of changing it later.